ros/src/twist_controller/dbw_node.py

```diff
@@ -32,66 +39,120 @@
-        wheel_radius = rospy.get_param('~wheel_radius', 0.2413)
-        wheel_base = rospy.get_param('~wheel_base', 2.8498)
-        steer_ratio = rospy.get_param('~steer_ratio', 14.8)
-        max_lat_accel = rospy.get_param('~max_lat_accel', 3.)
-        max_steer_angle = rospy.get_param('~max_steer_angle', 8.)
+        self.vehicle_mass = rospy.get_param('~vehicle_mass', 1736.35)
+        self.fuel_capacity = rospy.get_param('~fuel_capacity', 13.5)
+        self.brake_deadband = rospy.get_param('~brake_deadband', .1)
+        self.decel_limit = rospy.get_param('~decel_limit', -5)
+        self.accel_limit = rospy.get_param('~accel_limit', 1.)
+        self.wheel_radius = rospy.get_param('~wheel_radius', 0.2413)
+        self.wheel_base = rospy.get_param('~wheel_base', 2.8498)
+        self.max_lat_accel = rospy.get_param('~max_lat_accel', 3.)
+
 
         self.steer_pub = rospy.Publisher('/vehicle/steering_cmd',
-                                         SteeringCmd, queue_size=1)
+                                         SteeringCmd, queue_size=10)
         self.throttle_pub = rospy.Publisher('/vehicle/throttle_cmd',
-                                            ThrottleCmd, queue_size=1)
+                                            ThrottleCmd, queue_size=10)
         self.brake_pub = rospy.Publisher('/vehicle/brake_cmd',
-                                         BrakeCmd, queue_size=1)
+                                         BrakeCmd, queue_size=10)
 
         # TODO: Create `TwistController` object
-        # self.controller = TwistController(<Arguments you wish to provide>)
+        steer_ratio = rospy.get_param('~steer_ratio', 14.8)
+        max_steer_angle = rospy.get_param('~max_steer_angle', 8.)
+
+        self.yaw_controller =  YawController(self.wheel_base, steer_ratio, 0, self.max_lat_accel, max_steer_angle)
 
-        # TODO: Subscribe to all the topics you need to
+        rospy.Subscriber('/vehicle/dbw_enabled', Bool, self.on_enabled)
+        rospy.Subscriber('/twist_cmd_t', TwistStamped, self.on_twist_cmd)
+        rospy.Subscriber('/current_velocity', TwistStamped, self.on_current_velocity)
 
+        rospy.Subscriber('/throttle_pid/control_effort', Float64, self.on_control_effort)
+        self.pid_enable_pub = rospy.Publisher('/throttle_pid/enable', Bool, queue_size=1)
+        self.pid_state = rospy.Publisher('/throttle_pid/state', Float64, queue_size=1)
+        self.pid_setpoint = rospy.Publisher('/throttle_pid/setpoint', Float64, queue_size=1)
         self.loop()
 
+    def on_control_effort(self, data):
+        throttle = data.data
+        tcmd = ThrottleCmd()
+        tcmd.pedal_cmd_type = ThrottleCmd.CMD_PERCENT
+
+        bcmd = BrakeCmd()
+        #torque = Mass * acc * wheel_radius
+        max_brake_torque = min(BrakeCmd.TORQUE_MAX, (self.vehicle_mass + self.fuel_capacity * GAS_DENSITY)
+                               * abs(self.decel_limit) * self.wheel_radius)
+
+        if throttle >= 0:
```

Comment:
If you're using a deadband, then why is the throttle engaged if greater than 0? Shouldn't it only be engaged if greater than the deadband?